core/src/main/scala/com/databricks/labs/remorph/generators/sql/ExpressionGenerator.scala

```diff
@@ -11,7 +11,7 @@ import java.util.Locale
 class ExpressionGenerator(val callMapper: ir.CallMapper = new ir.CallMapper())
     extends Generator[ir.Expression, String] {
   private val dateFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd")
-  private val timeFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS").withZone(ZoneId.of("UTC"))
+  private val timeFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss").withZone(ZoneId.of("UTC"))
```

Comment:
so, actually we have to supply these based on the source dialect. e.g. for databricks there's one default format, for snowflake - another, for tsql - another, etc etc.

Reply:
You're right, but this `timeFormat` is only ever used for printing literals of the form `Literal(timestamp = Some(secondsSinceEpoch))` as DB-compliant timestamp strings. That is, for example, to generate `CAST('2024-08-20 18:09:00' AS TIMESTAMP)` from `Literal(timestamp = Some(17238293482397))`. 
The dialect-specific conversion you mention happens on the parser side, when we use the input string to build a `Literal(timestamp = Some(secondsSinceEpoch))`. 

That being said, I made this change so that my output timestamp literals match with the ones in the examples under `tests/`. When we'll be comparing the results our transpiled query actually produce against the results of the corresponding input queries, we'll have to handle precision much more carefully (as deciding whether we should print out milliseconds or not), which I expect to be a real _casse-tête_...